homeassistant/components/madvr/sensor.py

```diff
@@ -0,0 +1,239 @@
+
+
+def is_valid_temperature(value: float | None) -> bool:
+    """Check if the temperature value is valid."""
+    return value is not None and value > 0
+
+
+def get_temperature(coordinator: MadVRCoordinator, key: str) -> float | None:
+    """Get temperature value if valid, otherwise return None."""
+    temp = coordinator.data.get(key)
+    return float(str(temp)) if is_valid_temperature(temp) else None
+
+
+@dataclass(frozen=True, kw_only=True)
+class MadvrSensorEntityDescription(SensorEntityDescription):
+    """Describe madVR sensor entity."""
+
+    value_fn: Callable[[MadVRCoordinator], float | str | None]
+
+
+SENSORS: tuple[MadvrSensorEntityDescription, ...] = (
+    MadvrSensorEntityDescription(
+        key="mac_address",
+        name="MAC Address",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda coordinator: coordinator.data.get("mac_address"),
+        translation_key="mac_address",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_gpu",
+        name="GPU Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_gpu"),
+        translation_key="temp_gpu",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_hdmi",
+        name="HDMI Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_hdmi"),
+        translation_key="temp_hdmi",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_cpu",
+        name="CPU Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_cpu"),
+        translation_key="temp_cpu",
+    ),
+    MadvrSensorEntityDescription(
+        key="temp_mainboard",
+        name="Mainboard Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda coordinator: get_temperature(coordinator, "temp_mainboard"),
+        translation_key="temp_mainboard",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_res",
+        name="Incoming Resolution",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_res"),
+        translation_key="incoming_res",
+    ),
+    MadvrSensorEntityDescription(
+        key="incoming_frame_rate",
+        name="Incoming Frame Rate",
+        value_fn=lambda coordinator: coordinator.data.get("incoming_frame_rate"),
+        translation_key="incoming_frame_rate",
+    ),
```

Comment:
We can add a frequency here

Reply:
It reports it as frame rate not refresh rate - 23.976p

I can convert it, but seems hacky IMO because 23.976p has a specific connotation besides refresh rate (its a progressive signal)